src/components/experiment-tracking/experiment-primary-toolbar/experiment-primary-toolbar.js

```diff
@@ -26,6 +31,8 @@ export const ExperimentPrimaryToolbar = ({
     });
   };
 
+  const exportData = constructExportData(runMetadata, runTrackingData);
```

Comment:
This can be lazy as `CSVLink` accepts a function as well.

Also this can be a good candidate for `useCallback` (or if you want to stay with a static version of the data instead of a function, `useMemo`) so that the function is only re-created when dependencies change, and I'd suggest making runs selection as a dependency, as from the look of it, the requirement is only export data from selected run.

Reply:
Agree, and related to my comment [here](https://github.com/kedro-org/kedro-viz/pull/757#pullrequestreview-892303855).

Reply:
Thanks for the recommendations both! very valid points!

I had a look into using `useCallback` but unfortunately it doesn't work with the setup of `CSVLink` as it would generally take in the value of the object that was passed in on initialization. That said, they do have a `asyncOnClick` prop that seems to allow update of the export data, which would then allow the data object passed in to be updated via the `onClick` callback. 

Regardless, I can look into refactoring this and might put in a small follow up PR for this. Will tag you once it's out.